lib/srv/usermgmt_linux.go

```diff
@@ -71,3 +75,31 @@ func (*HostUsersProvisioningBackend) DeleteUser(name string) error {
 	}
 	return trace.Wrap(err)
 }
+
+func (*HostUsersProvisioningBackend) TestSudoersFile(contents []byte) error {
+	code, err := host.TestSudoersFile(contents)
+	if err != nil {
+		if code != 0 {
+			return trace.Errorf("visudo: invalid sudoers file")
+		}
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+func (u *HostUsersProvisioningBackend) WriteSudoersFile(username string, contents []byte) error {
+	sudoersFilePath := filepath.Join(u.sudoersPath, fmt.Sprintf("%s-%s", "teleport", username))
+	err := os.WriteFile(sudoersFilePath, contents, 0440)
```

Comment:
We should create a temporary file in `sudoersPath` with a `.` in the name and then rename it, as per `sudoers(5)`:
```
     sudo will suspend processing of the current file and read each file
     in /etc/sudoers.d, skipping file names that end in ‘~’ or contain a
     ‘.’ character to avoid causing problems with package manager or
     editor temporary/backup files.
```

For peace of mind it ought to be a temporary file created (with `O_EXCL`) as with `os.CreateTemp` or similar, so we are guaranteed that we're only ever dropping fully formed, valid sudoers files in there.

Reply:
should probably check if the file exists before writing it as well